RELEASE.md

```diff
@@ -1,3 +1,71 @@
+# oqs-provider 0.7.1
```

Comment:
@praveksharma @SWilson4 @zadlg imo wrong version id, don't you think (albeit it seems intentional, so maybe I'm wrong?)? Sorry for only looking at/asking this a day after the PR was created (and for not catching it already in rc1 -- probably just "pattern matched -rc1", mea culpa). 

Suggest adding wording to [release docs](https://github.com/open-quantum-safe/oqs-provider/wiki/Release-process) to ascertain release id in documentation fits actual release id chosen. That said, had one followed the current wording there

> Copy full release description to the top of RELEASES.md

the error (if you'd agree it is one) would not have occurred as the version in the release statement is correct.

Either way, something needs to be improved (release doc be brought in line with release procedure or release procedure itself be improved).

Reply:
Oh dear, that's embarrassing. :grimacing: This slipped past quite a few pairs of eyes, both here and in https://github.com/open-quantum-safe/oqs-provider/pull/533. I wonder if it warrants some sort of short "sanity check" script to run as part of release tests (i.e., just check that the version in CMakeLists matches the version in this file).

Reply:
> To correct this I can change the tags, force push, and updated the release.

This indeed is now possible again thanks to the relaxed GH access permissions. I struggle a bit with the implications as this basically re-writes "history"... Personally, I'd do it, though -- if only to avoid (possibly many future) questions such as to "which version is which now".

> The plan to mitigate such developer errors with releases going forward is to automate the release process using GitHub Actions and include sanity checks as suggested by @SWilson4. 

That would be nice, indeed.

Reply:
Oh, and going forward, I'd recommend a bit more time between PR and merge -- unless/until there's a rock-solid, fully automated release process in place.

Reply:
Before doing a rewrite of history, perhaps we ought to create and push a new branch from the current 0.7.0 and push it. That preserves the SHA for the commit originally tagged as 0.7.0, on the off chance that somebody out there is pulling by SHA instead of by tag.

Reply:
> That preserves the SHA (albeit not on the `main` branch) for the commit originally tagged as 0.7.0, on the off chance that somebody out there is pulling by SHA instead of by tag.

I tested this out in a fork and going about it as follows will move the 0.7.0 tag to a new commit while preserving the old commit hash so pulling by hash doesn't break:

```
git checkout 0.7.0-release
# Update release notes and commit changes.
git tag -f 0.7.0
git push
git push origin -f 0.7.0
```

Reply:
> Oh, and going forward, I'd recommend a bit more time between PR and merge -- unless/until there's a rock-solid, fully automated release process in place.

Certainly @baentsch, I will keep this mind. I can update the release wiki to require soliticiting at least X reviews or waiting at least Y days before merging the PR. Any thoughts on what are suitable values for X and Y?

Reply:
Well, for `liboqs` X and Y seem to be a week. I know, makes for a long overall release cycle, but the only alternative is more automation or more people working on the project in a dedicated manner thus guaranteeing timely reviews.